components/utils.py

```diff
@@ -810,6 +822,11 @@ def _get_upgrade_properties(self, service_name):
         with open(upgrade_props_file) as f:
             return json.load(f)
 
+    # XXX reconcile these two with the definitions above
+    def get_install_properties(self, service_name):
```

Comment:
We get the original inputs - from the `._get_props_file_path` file, and the upgrade inputs - from the `._load_ctx_properties` - is that right? Note that this pre-validation runs before the rollback etc files are created.
Should we use this method?